core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -0,0 +1,114 @@
+package org.infinispan.expiration.impl;
+
+import net.jcip.annotations.ThreadSafe;
+import org.infinispan.AdvancedCache;
+import org.infinispan.commands.CommandsFactory;
+import org.infinispan.commands.write.RemoveExpiredCommand;
+import org.infinispan.commons.util.Util;
+import org.infinispan.container.entries.ExpiryHelper;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.context.InvocationContext;
+import org.infinispan.context.InvocationContextContainer;
+import org.infinispan.context.InvocationContextFactory;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.interceptors.InterceptorChain;
+
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@ThreadSafe
+public class ClusterExpirationManager<K, V> extends ExpirationManagerImpl<K, V> {
+
+   private ExecutorService asyncExecutor;
+   private AdvancedCache<K, V> cache;
+
+   @Inject
+   public void inject(AdvancedCache<K, V> cache,
+           @ComponentName(KnownComponentNames.ASYNC_OPERATIONS_EXECUTOR) ExecutorService asyncExecutor) {
+      this.cache = cache;
```

Comment:
I still think you could have cast to `CacheImpl` here to avoid exposing `removeExpired` in `AdvancedCache`...